src/views/templates/details/hooks/useIsTemplateEditable.ts

```diff
@@ -0,0 +1,32 @@
+import { TemplateModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { useAccessReview } from '@openshift-console/dynamic-plugin-sdk';
+
+import { isCommonVMTemplate } from '../../utils/utils';
+
+/**
+ * This hook checks the user capabilities of editing templates in a namespace.
+ * Checks user permissions and template characteristics.
+ * The hook does not check if the user has permissions for that particular template
+ * as this behavior would compromise performarce in pages were a lot of templates are involved like template list.
+ * @param template template to check
+ * @returns boolean value
+ */
+const useIsTemplateEditable = (template: V1Template) => {
+  const [canUpdateTemplate, canUpdateLoading] = useAccessReview({
+    verb: 'update',
+    resource: TemplateModel.plural,
+    namespace: template?.metadata?.namespace,
+  });
+
+  const [canPatchTemplate, canPatchLoading] = useAccessReview({
+    verb: 'patch',
+    resource: TemplateModel.plural,
+    namespace: template?.metadata?.namespace,
+  });
+
+  if (!template || canUpdateLoading || canPatchLoading) return false;
+
+  return canUpdateTemplate && canPatchTemplate && !isCommonVMTemplate(template);
+};
+
```

Comment:
no loading or error return?

Reply:
`useAccessReview` has only `loading` state and does not have an error state. In case of an error, the hook will assume the user has permission. In a bad scenario, the server will stop the request anyways. I can expose the loading state but i don't use it in components